File: data_structures/arrays/static_array.py
from algorithms.searching.linear_search.linear_search import linear_search


class StaticArray:
  # On static arrays, the size of it is declared at the time
  # the arrays is created and can not be changed afterwards
  def __init__(self, length):
    self.length = length
    self.last_position = -1
    self.values = [None] * length

  # O(1)
  def insert(self, index, value):
      if index >= self.length:
          raise IndexError('index out of bond')
      self.values[index] = value
      return self

  # O(1)
  def delete(self, index):
      if index >= self.length:
          raise IndexError('index out of bond')
      self.values[index] = None
      return self

  # O(n)
  def linear_search(self, value):
    return linear_search(self.values, value)

File: data_structures/arrays/static_array.js
const {linearSearch} = require('../../algorithms/searching/linear_search/linear_search.js')


class StaticArray {
    constructor(length) {
        this.length = length;
        this.values = new Array(length).fill(null);
    }

    // O(1)
    insert(index, value) {
        if (index >= this.length) {
          return null;
        }
        this.values[index] = value;
        return this;
    }
      
    // O(1)
    delete(index) {
        if (index >= this.length) {
            return null;
        }
        this.values[index] = null;
        return this;
    }

    linearSearch(value) {
        return linearSearch(this.values, value);
    }
}

